pkg/controllers/provisioning/scheduling/reservationmanager.go

```diff
@@ -90,3 +96,15 @@ func (rm *ReservationManager) Release(hostname string, offerings ...*cloudprovid
 		}
 	}
 }
+
+func (rm *ReservationManager) HasReservation(hostname string, offering *cloudprovider.Offering) bool {
```

Comment:
nit: Can you use this function in the `Reserve()` call since this has the same logic? I get you may have to re-check the map, so I could go either way on it